pkg/server/cloud/aws/instances.go

```diff
@@ -462,23 +466,25 @@ func (e *AwsEC2) isHostAvailable(hostId *string) bool {
 		return false
 	}
 	host := describeOutput.Hosts[0]
-	return *host.State == "available"
+	return aws.StringValue(host.State) == AwsInstanceAvailableState
 }
 
-func (e *AwsEC2) waitForHostAvailable(ctx context.Context, hostId *string) error {
+func (e *AwsEC2) waitForHostAvailable(ctx context.Context, hostId *string) bool {
 	hostAvailable := false
+	ticker := time.NewTicker(1*time.Second)
+	quit := make(chan bool)
 	for {
 		select {
 		case <-ctx.Done():
-			if !hostAvailable {
-				return fmt.Errorf("host not available")
-			} else {
-				return nil
-			}
-		default:
+			return hostAvailable
+		case <-ticker.C:
 			hostAvailable = e.isHostAvailable(hostId)
+			if hostAvailable {
+				quit <- true
+			}
+		case <-quit:
+			return hostAvailable
```

Comment:
This could be simplified further if you removed `quit`, and return `hostAvailable` directly when it's true after the `isHostAvailable()` check.